test-two/src/helpers/http.js

```diff
@@ -1,18 +1,55 @@
 import fetch from "node-fetch";
 
-export const get = async url => {
+export const request = async (url, requestInit = {}) => {
   const response = await fetch(url, {
-    method: "GET",
-    headers: {
-      "User-Agent": "request"
-    }
+    ...requestInit,
   });
 
   const json = await response.json();
 
   if (response.status === 200) {
     return json;
   } else {
-    throw Error(json.Message);
+    const error = json.message || json.Message || json;
+    throw Error(error);
   }
+}
+
+export const get = async (url, requestInit = {}) => {
+  return await request(url, {
+    ...requestInit,
+    method: "GET"
+  });
 };
+
+export const post = async (url, requestInit = {}) => {
+  return await request(url, {
+    ...requestInit,
+    method: "POST"
+  });
+}
+
+export const joinPaginatedRequests = async (
+  total,
+  pageSize,
+  getter,
+) => {
+  let i = 0;
+  let maxPages = Math.ceil(total / pageSize);
+  const requests = [];
+
+  for (let index = 0; index < maxPages; index++) {
+    requests.push(
+      getter(pageSize, index)
+    );
+  }
+
+  const results = await Promise.all(requests);
+
+  return results
+    .filter(result => result)
+    .reduce((prev, curr) => [
+      ...prev,
+      ...curr
+    ], []);
+};
```

Comment:
Can you explain why you've taken this approach with the `filter` and `reduce`?

Reply:
Sorry for the late reply.
* filter:
 - The filter is to avoid empty or undefined arrays, but while I was thinking on the response, I reckon that a `filter(result => result && result.length)` would be more accurate, as it would be sure is an array and has elements, so we avoid unnecessary array concatenation.
* reduce:
 - the reduce was my first guess, given how useful MapReduce pattern is, but in this case I reckon a simple flatMap(result => result) serves the same purpose, I am not sure, I didnt measure it, but I think it might be more efficient even... (I know that in c#, Java and Rust that would be, given it uses iterables/stream, but js for that one still recreates arrays at each step, but maybe the later transpilers might be a bit smarter and optimize it? I don't know)